module/preference/PreferenceDialog.kt

```diff
@@ -453,8 +482,12 @@ object PreferenceDialog {
     @Composable
     private fun Preferences() {
         state.rootPreferenceGroup.resetSelfAndDescendants()
+        selectedPreferenceGroup = state.rootPreferenceGroup.entries.first()
```

Comment:
There are three ways a dialog can be closed:
1. Programatically, i.e. dialog.close()
2. Through the OS UI close button, e.g. the left-most red dot.
3. By pressing escape.

Dialog provides an `onCloseRequest` lambda argument that we can add code to, but that doesn't trigger on pressing escape. 

Given there are many ways to close a dialog and we don't seem to be able to set code to run in the event of any of them, it's cleaner to just reset the state of the dialog as it opens.

Reply:
This incidentally probably works because we happen to be in a `Dialog`, but in the general case this code would be run on every recomposition of `Preferences`, meaning it doesn't _necessarily_ do what we say here.

Compose provides a means of running a block of code on the composition of an object via `LaunchedEffect`. I'd say we should use that here.

Reply:
Done.